samples/image_diff.cc

```diff
@@ -52,18 +64,28 @@ bool ParseArgs(const std::vector<std::string>& args, Options* opts) {
       return true;
     } else if (arg == "--rmse") {
       opts->compare_algorithm = CompareAlgorithm::kRMSE;
+      ++i;
+      if (i >= args.size()) {
+        std::cerr << "Missing tolerance value for RMSE comparison."
+                  << std::endl;
+        return false;
+      }
+      opts->tolerance = std::stof(std::string(args[i]));
```

Comment:
This won't catch an issue where user has forgotten to specify the tolerance value. If we have `--rmse image1.png` it would try to read image1.png as a number and then crash when trying to throw an exception. Compile options forbid the use of catch here.

Reply:
Yeah, sscanf seems to work nicely.